### PR TITLE
#863 core should send client a list of available languages on startup

### PR DESCRIPTION
**Pull request by daubaris**
#863 

Added the implementation of available languages.

Things to note:
Not quite sure if line 197 in tabs.rs is a good solution, would love to receive some feedback about it.

### CODE REVIEW

**Comment by daubaris** [comment]
Going to fix rpc tests

File: rust/core-lib/src/syntax.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub struct LanguageDefinition {
 /// A repository of all loaded `LanguageDefinition`s.
 #[derive(Debug, Default)]
 pub struct Languages {
-    named: HashMap<LanguageId, Arc<LanguageDefinition>>,
+    pub named: HashMap<LanguageId, Arc<LanguageDefinition>>,
```
Comment (cmyr) [inline rust/core-lib/src/syntax.rs]: this is unnecessary; `Languages` has an `iter` fn, which returns an iterator over the `LanguageDefinition` objects.

File: rust/core-lib/src/tabs.rs [hunk 2]
```diff
@@ -194,6 +194,8 @@ impl CoreState {
         let plugin_paths = self.config_manager.get_plugin_paths();
         self.plugins.reload_from_paths(&plugin_paths);
         let languages = self.plugins.make_languages_map();
+        let languages_ids = languages.named.keys().cloned().collect::<Vec<_>>();
```
Comment (cmyr) [inline rust/core-lib/src/tabs.rs]: as mentioned above, this can just be `languages.iter().cloned().collect::<Vec<_>>();`.

Comment (cmyr) [reply rust/core-lib/src/tabs.rs]: er, the above is slightly wrong. I would do this as,
```rust
langauges.iter().map(|l| l.name.clone()).collect::<Vec<_>>();
```

Comment (daubaris) [reply rust/core-lib/src/tabs.rs]: Great! I’ll fix it as soon as I get back from work!

**Pull request merged**